bundles/io/org.openhab.io.transport.mqtt/src/main/java/org/openhab/io/transport/mqtt/MqttService.java

```diff
@@ -91,7 +91,13 @@ public void updated(Dictionary<String, ?> properties) throws ConfigurationExcept
             } else if (property.equals("async")) {
                 conn.setAsync(Boolean.parseBoolean(value));
             } else if (property.equals("clientId")) {
+                if (value.length() > 65535) {
+                    logger.warn("clientId must be less than 65536 characters long");
+                    throw new ConfigurationException("clientId", "clientId must be less than 65536 characters long");
+                }
                 conn.setClientId(value);
+            } else if (property.equals("allowLongerClientIds")) {
+                conn.setAllowLongerClientIds(Boolean.parseBoolean(value));
```

Comment:
I think I know why -- the IO transport bundle is part of the core set of bundles, so you would have to find and remove the built-in one, and drop its replacement in the addons folder like you do with binding JAR files.